### PR TITLE
LOOP-2820 Change expiration reminder selector to only allow pod-allowed dates

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-2820

* Update expiration reminder date picker to choose fixed set of allowed dates
* Do not allow editing of expiration reminder after it has occurred
* Only allow scheduling of future expiration reminder dates.
* Update RoundedCard styling to work on iPod touch with non-inset behavior.

### CODE REVIEW

File: DashKit/DashPumpManager.swift [hunk 1]
```diff
@@ -625,22 +623,31 @@ open class DashPumpManager: PumpManager {
...
-    public var allowedExpirationReminderDateRange: ClosedRange<Date>? {
+    public var allowedExpirationReminderDates: [Date]? {
         guard let expiration = podExpiresAt else {
             return nil
         }
         
-        let earliest = expiration.addingTimeInterval(.hours(-24))
-        let latest = expiration.addingTimeInterval(.hours(-1))
-        return earliest...latest
+        let allDates = Array(stride(from: 24, through: 1, by: -1)).map { (i: Int) -> Date in
```
Comment (darinkrauss) [inline DashKit/DashPumpManager.swift]: [nit] Consider `from: -24, through: -1, by: 1` and getting rid of the `-i` on the line below?

File: DashKitUI/Views/NotificationSettingsView.swift [hunk 2]
```diff
@@ -40,29 +40,44 @@ struct NotificationSettingsView: View {
...
+            }
+            .padding(.horizontal, 16)
         }
         .navigationBarTitle(LocalizedString("Notification Settings", comment: "navigation title for notification settings"))
     }
     
+    var expirationDefaultFormatter = QuantityFormatter(for: .hour())
+    
+    var expirationDefaultString: String {
+        return expirationDefaultFormatter.string(from: HKQuantity(unit: .hour(), doubleValue: Double(expirationReminderDefault)), for: .hour())!
```
Comment (darinkrauss) [inline DashKitUI/Views/NotificationSettingsView.swift]: [nit] Using HealthKit to format a duration seems odd.

File: DashKitUI/Views/ScheduledExpirationReminderEditView.swift [hunk 3]
```diff
@@ -39,14 +45,25 @@ struct ScheduledExpirationReminderEditView: View {
...
-                RoundedCard {
-                    RoundedCardValueRow(
-                        label: LocalizedString("Scheduled Reminder", comment: "Label for scheduled expiration reminder row"),
-                        value: dateFormatter.string(from: selectedDate),
-                        highlightValue: true
-                    )
-                    DatePicker("", selection: $selectedDate, in: allowedReminderDateRange)
-                        .datePickerStyle(WheelDatePickerStyle())
+                if self.horizontalSizeClass == .compact {
+                    // Keep picker outside of card in compact view, because it
```
Comment (darinkrauss) [inline DashKitUI/Views/ScheduledExpirationReminderEditView.swift]: [nit] "because it" what?

**Review by darinkrauss — APPROVED** [review]
A couple nits, but LGTM!

File: DashKitUI/Views/NotificationSettingsView.swift (hunk 2 above)
Comment (ps2) [reply DashKitUI/Views/NotificationSettingsView.swift]: Hour units translations should be handled by the system.

File: DashKitUI/Views/ScheduledExpirationReminderEditView.swift (hunk 3 above)
Comment (ps2) [reply DashKitUI/Views/ScheduledExpirationReminderEditView.swift]: ... forces full device width.  Will update, thanks.

File: DashKit/DashPumpManager.swift (hunk 1 above)
Comment (ps2) [reply DashKit/DashPumpManager.swift]: Seems fairly equivalent to me, but it sounds like you think it reads better that way, so I'll update it.

**Pull request merged**